### PR TITLE
SELF -> DBO

### PR DESCRIPTION
**Pull request by SergeyGaluzo**
It also removes not used RebuildIndex stored procedure.

### CODE REVIEW

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Operations/Import/SqlServerFhirDataBulkOperationTests.cs [hunk 1]
```diff
@@ -268,17 +268,17 @@ public async Task GivenUnclusteredIndexes_WhenRebuildDisableIndexes_ThenIndexSho
                 isDisabled = await GetIndexDisableStatus(index.indexName);
                 Assert.True(isDisabled);
 
-                await RebuildIndex(index.tableName, index.indexName, index.pageCompression);
-                isDisabled = await GetIndexDisableStatus(index.indexName);
-                Assert.False(isDisabled);
+                ////await RebuildIndex(index.tableName, index.indexName, index.pageCompression);
```
Comment (LTA-Thinking) [inline test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Operations/Import/SqlServerFhirDataBulkOperationTests.cs:271]: Can we just remove these lines if they aren't needed anymore?

File: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/SwitchPartitionsInAllTables.sql [hunk 2]
```diff
@@ -1,7 +1,6 @@
 ﻿--IF object_id('SwitchPartitionsInAllTables') IS NOT NULL DROP PROCEDURE dbo.SwitchPartitionsInAllTables
 GO
 CREATE PROCEDURE dbo.SwitchPartitionsInAllTables
-WITH EXECUTE AS SELF
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/SwitchPartitionsInAllTables.sql:4]: Why is dbo not needed here?

File: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/SwitchPartitionsOutAllTables.sql [hunk 3]
```diff
@@ -1,7 +1,6 @@
 ﻿--IF object_id('SwitchPartitionsOutAllTables') IS NOT NULL DROP PROCEDURE dbo.SwitchPartitionsOutAllTables
 GO
 CREATE PROCEDURE dbo.SwitchPartitionsOutAllTables @RebuildClustered bit
-WITH EXECUTE AS SELF
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/SwitchPartitionsOutAllTables.sql:4]: Why is dbo not needed here?

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Operations/Import/SqlServerFhirDataBulkOperationTests.cs (hunk 1 above)
Comment (SergeyGaluzo) [reply test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Operations/Import/SqlServerFhirDataBulkOperationTests.cs:271]: I will let Tong to comment.

File: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/SwitchPartitionsInAllTables.sql (hunk 2 above)
Comment (SergeyGaluzo) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/SwitchPartitionsInAllTables.sql:4]: Permissions from one stored proc to another are transferable. I don't think impersonation is needed.

File: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/SwitchPartitionsOutAllTables.sql (hunk 3 above)
Comment (SergeyGaluzo) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/SwitchPartitionsOutAllTables.sql:4]: Permissions from one stored proc to another are transferable. I don't think impersonation is needed.

**Review by LTA-Thinking — APPROVED** [review]

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Operations/Import/SqlServerFhirDataBulkOperationTests.cs (hunk 1 above)
Comment (tongwu-msft) [reply test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Operations/Import/SqlServerFhirDataBulkOperationTests.cs:271]: We can remove old rebuild indexes SProc related code, they are not needed any more.

**Review by tongwu-msft — DISMISSED** [review]

**Review by LTA-Thinking — APPROVED** [review]

**Pull request merged**